src/components/VerboseTextField.tsx

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright 2022 Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+/* eslint-disable react/jsx-props-no-spreading */
+
+import TextField, { OutlinedTextFieldProps } from "@mui/material/TextField";
+import { FormHelperText, FormHelperTextProps, styled } from "@mui/material";
+
+export type FieldStatus = "error" | "warning" | "info" | "success" | undefined;
+
+export interface VerboseFieldState {
+  statusDescription: string;
+  statusValue: FieldStatus;
+}
+
+export interface FormTextFieldProps extends OutlinedTextFieldProps {
+  state?: VerboseFieldState;
+  description?: string;
+  formDescriptionTextProps?: FormHelperTextProps;
+}
+
+const StyledTextField = styled(TextField)<FormTextFieldProps>(({ theme }) => ({
+  "& .Mui-error": {
+    color: theme.palette.error.main,
+  },
+  "& .Mui-warning": {
+    color: theme.palette.warning.main,
+  },
+  "& .Mui-info": {
+    color: theme.palette.info.dark,
```

Comment:
Out of curiosity, why use the `dark` theme only for `info` and `main` for the other states?

Reply:
![image](https://user-images.githubusercontent.com/59744144/192250170-0baa3b22-8300-4eb3-952a-6a2c4ecbbe75.png)
![image](https://user-images.githubusercontent.com/59744144/192250234-591eb57c-0e74-4254-a086-5f30feaa424f.png)

I found the info.main color one a bit harder to read contrast-wise so I decided to go with the dark theme for this one.

Reply:
Should I change the theme instead?

Reply:
I would say so yes, maybe run this by Kyra I'm not sure what the best practice is in this case.

Reply:
I'll catch up with Kyra and open a separate PR for this.